Stacktraces fail when one session scoped fixture depending on another session scoped fixture crashes

```python
import pytest


@pytest.fixture(scope="session")
def first():
    return "first"


@pytest.fixture(scope="session")
async def second(first):
    assert False


@pytest.mark.asyncio_cooperative
async def test_hello(second):
    print("hello")
```


If these fixtures were not session scoped we would see:
```
    @pytest.fixture
    async def second(first):
>       assert False
E       assert False
```
However when these fixtures are declared to have a session scope we see:
```
in call_fixture_func
    fixture_result = fixturefunc(**kwargs)
E   TypeError: CachedFunction.__call__() got an unexpected keyword argument 'first'
```
Especially when working with complex and nested fixtures, it's valuable to be able to see the line that causes an error, rather than only an unrelated line belonging to pytest.

--- a/pytest_asyncio_cooperative/fixtures.py
+++ b/pytest_asyncio_cooperative/fixtures.py
@@ -159,8 +159,9 @@ class CachedGen(CachedFunctionBase):
     def completed(self, instance):
         self.instances.remove(instance)
 
-    def __call__(self, *args):
+    def __call__(self, *args, **kwargs):
         self.args = args
+        self.kwargs = kwargs
         instance = GenCounter(self)
         self.instances.add(instance)
         return instance
@@ -174,7 +175,7 @@ class CachedGen(CachedFunctionBase):
             raise self.exception
         else:
             try:
-                gen = self.wrapped_func(*self.args)
+                gen = self.wrapped_func(*self.args, **self.kwargs)
                 self.gen = gen
                 self.value = gen.__next__()
             except Exception as e:
@@ -209,8 +210,9 @@ class CachedAsyncGen(CachedFunctionBase):
     def completed(self, instance):
         self.instances.remove(instance)
 
-    def __call__(self, *args):
+    def __call__(self, *args, **kwargs):
         self.args = args
+        self.kwargs = kwargs
         instance = AsyncGenCounter(self)
         self.instances.add(instance)
         return instance
@@ -225,7 +227,7 @@ class CachedAsyncGen(CachedFunctionBase):
                 raise self.exception
             else:
                 try:
-                    gen = self.wrapped_func(*self.args)
+                    gen = self.wrapped_func(*self.args, **self.kwargs)
                     self.gen = gen
                     self.value = await gen.__anext__()
                 except Exception as e:
@@ -243,13 +245,14 @@ class CachedAsyncGenByArguments(CachedAsyncGen):
         super().__init__(wrapped_func)
         self.callers_by_args = {}
 
-    def __call__(self, *args):
+    def __call__(self, *args, **kwargs):
+        # TODO: Handle cases where args are the same, but kwargs differ
         if args in self.callers_by_args:
             gen = self.callers_by_args[args]
         else:
             gen = CachedAsyncGen(self.wrapped_func)
             self.callers_by_args[args] = gen
-        return gen(*args)
+        return gen(*args, **kwargs)
 
 
 async def _make_asyncgen_fixture(_fixtureinfo, fixture: FixtureDef, item):

--- a/tests/test_fixture.py
+++ b/tests/test_fixture.py
@@ -539,3 +539,30 @@ def test_getting_fixture_from_closest_conftest(testdir):
     )
     result = testdir.runpytest()
     result.assert_outcomes(passed=2)
+
+
+def test_fixture_nested_exception(testdir):
+    testdir.makepyfile(
+        f"""
+        import pytest
+
+
+        @pytest.fixture(scope="module")
+        async def first():
+            return "first"
+
+
+        @pytest.fixture(scope="module")
+        async def second(first):
+            assert False
+            yield first
+
+
+        @pytest.mark.asyncio_cooperative
+        async def test_hello(second):
+            print("hello")
+        """
+    )
+
+    result = testdir.runpytest()
+    result.assert_outcomes(errors=0, failed=1)

